File: Cc.Core/Models/CustomerRequiredField.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models
{
    public partial class CustomerRequiredField
    {
        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public string InsertUser { get; set; } = null!;
        public DateTime UpdateDate { get; set; }
        public string UpdateUser { get; set; } = null!;
        public long? CompanyId { get; set; }
        public long? AgencyId { get; set; }
        public long? ProductId { get; set; }
        public bool? Base { get; set; }
        public bool? Birth { get; set; }
        public bool? Identification { get; set; }

        public virtual Agency? Agency { get; set; }
        public virtual Company? Company { get; set; }
        public virtual Product? Product { get; set; }
    }
}

File: Cc.Ui/src/models/customerRequiredField.ts
import { Entity, resource, field, required } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';

@resource('customerRequiredField')
export class CustomerRequiredField extends Entity {

    @field()
    id: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    companyId: number;
    @field()
    agencyId: number;
    @field()
    productId: number;
    @field()
    base: boolean;
    @field()
    birth: boolean;
    @field()
    identification: boolean;

    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): CustomerRequiredField {

        let item = new CustomerRequiredField();
        return item;
    }
}

@resource('customerRequiredField')
export class CustomerRequiredFieldSearchModel extends Entity implements Pagination {
    
    @field()
    companyId?: number;
    @field()
    agencyId?: number;
    @field()
    productId?: number;
    @field()
    base?: boolean;
    @field()
    birth?: boolean;
    @field()
    identification?: boolean;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}